lib/wit/main.py

```diff
@@ -108,6 +111,37 @@ def main() -> None:
         raise WitBug(e)
 
 
+def foreach(ws, args):
+    packages = sorted(ws.lock.packages, key=lambda x: str.lower(x.name))
+
+    has_fail = False
+    for pkg in packages:
+        env = os.environ.copy()
+        env["WIT_REPO_NAME"] = pkg.name
+        env["WIT_REPO_PATH"] = str(ws.root / pkg.name)
+        env["WIT_LOCK_SOURCE"] = pkg.source
+        env["WIT_LOCK_COMMIT"] = pkg.revision
+        env["WIT_WORKSPACE"] = str(ws.root)
+
+        if not args.quiet:
+            print("Entering '{}'".format(pkg.name))
```

Comment:
Should this be governed by the logger? I'm wondering if the whole `--quiet` option should be folded into that then this print would just be `log.info`.

Reply:
I'm trying to copy the usage from `git submodule foreach` usage, 
from the git docs:
```Unless given --quiet, foreach prints the name of each submodule before evaluating the command. ```

Reply:
I agree with generally following git's lead, but since we have unified logging (although I think it currently lacks a mechanism to _lower_ the verbosity which we obviously need), I would think it would be better to keep it unified with the logging API. I think it would be strange if just 1 command ignores your logging settings.

Reply:
If we make a global `--quiet`, we'd need to do it properly and make all the `print()` in the codebase conditional rather than just the log lines.

How about if I make the print line in this snippet unconditional, but print the line to stderr so that people who don't want it can filter it out with 2>/dev/null ?

Reply:
though grep-ing, we don't actually have that many `print()`

Reply:
I'll break this stuff into a separate diff

Reply:
https://github.com/sifive/wit/pull/214